### ISSUE TITLE
Question about the output of the model

### ISSUE THREAD

**Issue by TrayfourYu** [qiyan98/TrajFlow#6]
Hi， thanks for the great job! I wonder the output of the model is normalized traj or not when the "objective == 'pred_data' " ?

according to the training code below， it should be the de-normalized traj. 
`       denoised_x_metric = pred_list[-1][1][..., :2]                                       # [B, K, T, 2]`
`        denoised_x = shift_data_to_normalize(denoised_x_metric, None, self.data_rescale) `

But by the sampling code, it should be the normalized traj. 
`        if flag_sample:`
`            # Generate trajectories`
`            trajs, cls = self.sample(batch_dict)`
`            trajs = shift_data_to_denormalize(trajs, None, data_rescale=self.data_rescal`

**Comment by smilebetterworld** [comment]
@qiyan98 @TrayfourYu ， I have the same doubt, and I think the following is wrong and it should be shift_data_to_denormalize, right?
 denoised_x = shift_data_to_normalize(denoised_x_metric, None, self.data_rescale)

**Comment by qiyan98** [comment]
Hi, thanks for your comments, and sorry for the late reply.

In the refactored code, we try to align with commonly adopted GMM-based regression loss implementations such as MTR, EDA, and BeTop.

Under our internal tests, this regression loss works best when the scale is in metric space rather than a normalized range. For simplicity, we let the neural network’s raw outputs represent metric-scale GMM parameters. This is why there are a few places in the code with such transformations.

We would like to note that this part is purely empirical and intended to achieve better performance. If we consistently stay in a normalized range and use simple L1 or L2 regression losses, training can still work with a proper reweighting scheme, as observed in our internal model development logs. However, the final performance is slightly weaker, which is why we chose the current implementation.
